internal/types/host/keepalived_host_collector_host.go

```diff
@@ -0,0 +1,270 @@
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		logrus.WithFields(logrus.Fields{"stderr": stderr.String(), "stdout": stdout.String()}).WithError(err).Error("Error getting keepalived version")
+		return nil, errors.New("Error getting keepalived version")
+	}
+
+	version, err := utils.ParseVersion(stdout.String())
+	if err != nil {
+		logrus.WithError(err).Error("Failed to parse version output")
+		return nil, errors.New("Error parsing version output")
+	}
+
+	return version, nil
+}
+
+// Signal sends signal to Keepalived process
+func (k *KeepalivedHostCollectorHost) signal(signal os.Signal) error {
+	data, err := ioutil.ReadFile(k.pidPath)
+	if err != nil {
+		logrus.WithField("path", k.pidPath).WithError(err).Error("Can't find keepalived")
+		return err
+	}
+
+	pid, err := strconv.Atoi(strings.TrimSuffix(string(data), "\n"))
+	if err != nil {
+		logrus.WithField("path", k.pidPath).WithError(err).Error("Unknown pid found for keepalived")
+		return err
+	}
+
+	proc, err := os.FindProcess(pid)
+	if err != nil {
+		logrus.WithField("pid", pid).WithError(err).Error("Failed to find process")
+		return err
+	}
+
+	err = proc.Signal(signal)
+	if err != nil {
+		logrus.WithField("pid", pid).WithError(err).Error("Failed to send signal")
+		return err
+	}
+
+	// Wait 10ms for Keepalived to create its files
+	time.Sleep(10 * time.Millisecond)
+	return nil
+}
+
+// SigNum returns signal number for given signal name
+func (k *KeepalivedHostCollectorHost) sigNum(sigString string) syscall.Signal {
+	if !k.HasSigNumSupport() {
+		switch sigString {
+		case "DATA":
+			return syscall.SIGUSR1
+		case "STATS":
+			return syscall.SIGUSR2
+		default:
+			logrus.WithField("signal", sigString).Fatal("Unsupported signal for your keepalived")
+		}
+	}
+
+	sigNumCommand := "keepalived --signum=" + sigString
+	cmd := exec.Command("bash", "-c", sigNumCommand)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		logrus.WithFields(logrus.Fields{"signal": sigString, "stderr": stderr.String()}).WithError(err).Fatal("Error getting signum")
+	}
+
+	signum, err := strconv.ParseInt(stdout.String(), 10, 32)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{"signal": sigString, "signum": stdout.String()}).WithError(err).Fatal("Error parsing signum result")
+	}
+
+	return syscall.Signal(signum)
+}
+
+func (k *KeepalivedHostCollectorHost) HasSigNumSupport() bool {
+	return k.version == nil || k.version.GreaterThanOrEqual(sigNumSupportedVersion)
+}
+
+func (k *KeepalivedHostCollectorHost) jsonVrrps() ([]collector.VRRP, error) {
+	err := k.signal(k.SIGJSON)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to send JSON signal to keepalived")
+		return nil, err
+	}
+
+	f, err := os.Open("/tmp/keepalived.json")
+	if err != nil {
+		logrus.WithError(err).Error("Failed to open /tmp/keepalived.json")
+		return nil, err
+	}
+	defer f.Close()
+
+	VRRPs, err := collector.ParseJSON(f)
```

Comment:
I think we could log this err in the `ParseJSON` function and simply return the result here
```suggestion
	return collector.ParseJSON(f)
```

Reply:
:+1: